### PR TITLE
New Case Study: "A Peer Review Bot for GitHub"

### PR DESCRIPTION
**Pull request by felixrieseberg**
To test out the new workflow, I'm happy to jump in first: @sedouard @ritazh - I'd love it if you could give my case study a read and comment with 'LGTM' if you like it!

### CODE REVIEW

**Comment by stephbu** [comment]
LGTM

File: _posts/2015-08-01-Peer-Review-Bot-Github.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+* If the bot is configured to automatically merge 'approved' pull requests, we go and ask GitHub to merge it. If the pull request isn't mergable, GitHub will naturally refuse. 
+
+#### Let's Check Out Some Code
+Under the hood, the bot is just a small Express-driven app with one endpoint: http://botaddress.com/pullrequests. A GET HTTP request will check the configured repo for open pull requests, while POST HTTP requests expects a GitHub-style JSON payload and handles only pull requests mentioned in it.
+
+Using Mike de Boer's excellent GitHub Node module, all we had to was to write a module [describing all the individual interactions with GitHub](https://github.com/felixrieseberg/peer-review-bot/blob/master/bot.js) as well as a [small router module describing the review flow mentioned above](https://github.com/felixrieseberg/peer-review-bot/blob/master/routes/pullrequest.js). The bot itself is started with a well-known [bin/www file](https://github.com/felixrieseberg/peer-review-bot/blob/master/bin/www), which [itself calls app.js to setup the HTTP middleware](https://github.com/felixrieseberg/peer-review-bot/blob/master/app.js).
```
Comment (sedouard) [inline _posts/2015-08-01-Peer-Review-Bot-Github.md]: ' all we had to was to write a module' - this reads a bit funny. How bout 'we only needed to write a module...'

Comment (sedouard) [inline _posts/2015-08-01-Peer-Review-Bot-Github.md]: `The bot itself is started with a well-known [bin/www file](https://github.com/felixrieseberg/peer-review-bot/blob/master/bin/www), which [itself calls app.js to setup the HTTP middleware](https://github.com/felixrieseberg/peer-review-bot/blob/master/app.js).`

Is fairly uninteresting to me. Most interesting and connecting to your previous sentence is that the main `app.js` registers the `peerreview` bot router for the `/peerreview` path.

File: _posts/2015-08-01-Peer-Review-Bot-Github.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Setup Your Own Bot!
+If you too would like to require peer reviews of pull requests in your repo, you're in luck:  Before we get started, let's do some required housekeeping:
+
+#### 1) Creating a GitHub User for the Bot
+The bot will need a GitHub user to interact with GitHub. You can just use your own account, but no good bot was ever written without a good robot pun. Head over to GitHub and create a new user, choosing from the many robots out there - Terminator, Tinman, Bender - choose wisely.
+
+While the bot works fine with a username/password combination, we recommend setting up an OAuth2 token. GitHub has API rate limits which are a little bit higher for OAuth-authenticated calls. So while you're at it, go a
```
Comment (sedouard) [inline _posts/2015-08-01-Peer-Review-Bot-Github.md]: `So while you're at it, go a` looks like you forgot some text here.

File: _posts/2015-08-01-Peer-Review-Bot-Github.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+* `labelReviewed` Name of the label indicating enough peer reviews
+* `labelNeedsreview` Name of the label indicating missing peer reviews
+* `reviewsNeeded` Number of reviews needed 
+* `instructionsComment` Comment posted by the bot when a new PR is opened - if you use `{reviewsNeeded}` in your comment, it'll automatically be replaced with the number of reviews needed
+* `mergeOnReview` (default: false) If set to true, the bot will automatically merge a PR as soon as it consideres it revieweed
+* `pullRequestsStatus` (default: open) Status of the pull requests to consider. Options are: all|open|closed
+* `oauth2token` If set, we'll use an OAuth token instead of the username/password combination to authenticate the bot
+* `excludeLabels` If set, the bot will automatically ignore PRs with those labels (format: `no-review i-hate-reviews`)
+
+Once configures, start the bot with `npm start` or `node ./bin/www`. Now that a robot is taking care of your peer review workflow, you human can do something more fun!
```
Comment (sedouard) [inline _posts/2015-08-01-Peer-Review-Bot-Github.md]: `your* human co do something more fun!`

File: _posts/2015-08-01-Peer-Review-Bot-Github.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+* `repo` Repository to watch (case-studies)
+* `botUser` Bot's GitHub username
+* `botPassword` Bot's Github password
+* `labelReviewed` Name of the label indicating enough peer reviews
+* `labelNeedsreview` Name of the label indicating missing peer reviews
+* `reviewsNeeded` Number of reviews needed 
+* `instructionsComment` Comment posted by the bot when a new PR is opened - if you use `{reviewsNeeded}` in your comment, it'll automatically be replaced with the number of reviews needed
+* `mergeOnReview` (default: false) If set to true, the bot will automatically merge a PR as soon as it consideres it revieweed
+* `pullRequestsStatus` (default: open) Status of the pull requests to consider. Options are: all|open|closed
+* `oauth2token` If set, we'll use an OAuth token instead of the username/password combination to authenticate the bot
```
Comment (sedouard) [inline _posts/2015-08-01-Peer-Review-Bot-Github.md]: Mention that you can generate a token using [this guide](https://help.github.com/articles/creating-an-access-token-for-command-line-use/)

File: _posts/2015-08-01-Peer-Review-Bot-Github.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+author-link: "http://www.felixrieseberg.com"
+#author-image: "{{site.baseurl}}/images/authors/felix.jpg"
+date:   2015-08-01 10:00:00
+categories: GitHub Automation Node
+color: "blue"
+#image: "{{ site.baseurl }}/images/imagename.png" #should be ~350px tall
+excerpt: We built a peer review bot for GitHub, automating review workflows for pull requests - so that humans can spend their time with more fun things.
+---
+
+We in Microsoft's Open Source team occasionally write case studies explaining our work with a bit more technical depth for both the outside world as well as our fellow Microsoft engineers. Naturally, you don't want to embarrass yourself, so before any case study is released, you have to get some peer reviews.
```
Comment (ritazh) [inline _posts/2015-08-01-Peer-Review-Bot-Github.md]: We, in Microsoft's Open Source team, occasionally write case studies to explain our work with a bit more technical depth for both external engineers as well as our fellow internal Microsoft engineers. Naturally, we don't want to embarrass ourselves. So before any case study is released, we have to get some peer reviews.

File: _posts/2015-08-01-Peer-Review-Bot-Github.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+#### 1) Creating a GitHub User for the Bot
+The bot will need a GitHub user to interact with GitHub. You can just use your own account, but no good bot was ever written without a good robot pun. Head over to GitHub and create a new user, choosing from the many robots out there - Terminator, Tinman, Bender - choose wisely.
+
+While the bot works fine with a username/password combination, we recommend setting up an OAuth2 token. GitHub has API rate limits which are a little bit higher for OAuth-authenticated calls. So while you're at it, go a 
+
+#### 2) Deploying the Bot
+If you're using Azure, setting up a bot takes about a minute. Simply head over to the [Peer Review Bot's repository](http://github.com/felixrieseberg/peer-review-bot) and click the 'Deploy to Azure' button. The Azure deployment wizard will automatically create a website for you; but also setup environment variables configuring the bot itself.
+
+If you're anywhere else, simply make sure that you're running Node (at least on 0.10.x). Then, setup configuration by either directly editing `config.js` or by setting environment variables. Here are your options:
```
Comment (ritazh) [inline _posts/2015-08-01-Peer-Review-Bot-Github.md]: ```If you're anywhere else, simply make sure that``` --- "If you're anywhere else, make sure you're..."

File: _posts/2015-08-01-Peer-Review-Bot-Github.md (hunk 3 above)
Comment (ritazh) [inline _posts/2015-08-01-Peer-Review-Bot-Github.md]: ```Once configures``` -- "Once configured..."

File: _posts/2015-08-01-Peer-Review-Bot-Github.md (hunk 6 above)
Comment (ritazh) [inline _posts/2015-08-01-Peer-Review-Bot-Github.md]: Should probably mention something about make sure you don't checkin the 'config.js' file into github...

File: _posts/2015-08-01-Peer-Review-Bot-Github.md (hunk 4 above)
Comment (ritazh) [inline _posts/2015-08-01-Peer-Review-Bot-Github.md]: Should probably mention something about "If you add the OAuth token to your config.js file, make sure you don't checkin the 'config.js' file into GitHub".

**Comment by felixrieseberg** [comment]
@ritazh @sedouard: Thank you very much, I added all your feedback points. If you comment the magic phrase, I can merge this ;-)

**Comment by sedouard** [comment]
Why did the checker fail?

**Comment by felixrieseberg** [comment]
Well, Travis thought that dev.office.com is offline (which it maybe was). It passed on re-run though.

**Pull request merged**